run_flax_speech_recognition_seq2seq_streaming_review.py

```diff
@@ -0,0 +1,1391 @@
+
+    # Define generation function
+    num_beams = model_args.num_beams if model_args.num_beams is not None else model.config.num_beams
+    gen_kwargs = {"max_length": max_label_length, "num_beams": num_beams}
+
+     
+    def generate_step(params, batch):
+        model.params = params
+        attention_mask = batch.get("attention_mask")
+        output_ids = model.generate(batch[model_input_name], attention_mask=attention_mask, **gen_kwargs)
+        return output_ids.sequences
+
+    # Create parallel version of the train and eval step
+    p_train_step = jax.pmap(
+        partial(train_step, label_smoothing_factor=training_args.label_smoothing_factor), "batch", donate_argnums=(0, )
+    )
+    p_eval_step = jax.pmap(partial(
+        eval_step, label_smoothing_factor=training_args.label_smoothing_factor), "batch")
+    p_generate_step = jax.pmap(generate_step, "batch")
+
+    # Replicate the train state on each device
+    state = state.replicate()
+    
+    # Logging
+    logger.info("***** Running training *****")
+    logger.info(
+        f"  Original model = {model_args.model_name_or_path}")
+    if training_args.push_to_hub:
+        logger.info(
+        f"  Hub model id = {training_args.hub_model_id}")
+    logger.info(
+        f"  Dataset name = {data_args.dataset_name}")
+    logger.info(
+        f"  Dataset config name = {data_args.dataset_config_name}")
+    logger.info(
+        f"  Learning rate = {training_args.learning_rate}")
+    logger.info(
+        f"  Scheduler = {training_args.lr_scheduler_type}")
+    logger.info(
+        f"  Num examples = {data_args.num_train_steps * train_batch_size:,}")
+    if model_args.num_beams:
+        logger.info(
+        f"  Num beams evaluation = {model_args.num_beams}")
+    logger.info(
+        f"  Number of hosts = {num_of_hosts}")
+    if num_of_hosts > 1:
+        logger.info(
+            f"  Current host idx = {current_host_idx}")
+    logger.info(
+        f"  Instantaneous batch size per device = {training_args.per_device_train_batch_size}")
+    logger.info(
+        f"  Total train batch size per node (w. parallel & distributed) = {train_batch_size // num_of_hosts:,}")
+    logger.info(
+        f"  Total train batch size (w. parallel & distributed) = {train_batch_size}")
+    if training_args.gradient_accumulation_steps > 1:
+        logger.info(
+            f"  Gradient accumulation steps = {training_args.gradient_accumulation_steps}")
+        logger.info(f"  ↪ Effective total batch size = {train_batch_size * training_args.gradient_accumulation_steps:,}")
+    logger.info(f"  Total optimization steps = {data_args.num_train_steps - training_state['step']:,}")
+    if training_state['step'] > 0:
+        logger.info(f"  ↪ Starting at {training_state['step']:,} and finishing at {data_args.num_train_steps:,}")
+
+    if model_args.dropout or model_args.attention_dropout or model_args.activation_dropout or model_args.encoder_dropout or model_args.decoder_dropout:
```

Comment:
Again, this is pretty Whisper specific (e.g. Speech Encoder Decoder Models won't have these config attributes https://github.com/huggingface/transformers/tree/main/examples/pytorch/speech-recognition#warm-started-speech-encoder-decoder-model)

Reply:
Agree. Let's see where the script ends up being. Happy to remove all the dropout options.